pyotgw/pyotgw.py

```diff
@@ -358,53 +358,84 @@ async def get_status(self):
-            v.DATA_DHW_BURNER_STARTS: int(fields[20]),
-            v.DATA_TOTAL_BURNER_HOURS: int(fields[21]),
-            v.DATA_CH_PUMP_HOURS: int(fields[22]),
-            v.DATA_DHW_PUMP_HOURS: int(fields[23]),
-            v.DATA_DHW_BURNER_HOURS: int(fields[24]),
-        }
+        if len(fields) == 34:
+            # OpenTherm Gateway 5.0
+            capmodlimits = fields[6].split("/")
+            dhw_setp_bounds = fields[19].split("/")
+            ch_setp_bounds = fields[20].split("/")
+            status = {
+                v.DATA_SLAVE_MAX_RELATIVE_MOD: float(fields[5]),
+                v.DATA_ROOM_SETPOINT: float(fields[7]),
+                v.DATA_REL_MOD_LEVEL: float(fields[8]),
+                v.DATA_CH_WATER_PRESS: float(fields[9]),
+                v.DATA_ROOM_TEMP: float(fields[12]),
+                v.DATA_CH_WATER_TEMP: float(fields[13]),
+                v.DATA_DHW_TEMP: float(fields[14]),
+                v.DATA_OUTSIDE_TEMP: float(fields[15]),
+                v.DATA_RETURN_WATER_TEMP: float(fields[16]),
+                v.DATA_DHW_SETPOINT: float(fields[21]),
+                v.DATA_MAX_CH_SETPOINT: float(fields[22]),
+                v.DATA_TOTAL_BURNER_STARTS: int(fields[26]),
+                v.DATA_CH_PUMP_STARTS: int(fields[27]),
+                v.DATA_DHW_PUMP_STARTS: int(fields[28]),
+                v.DATA_DHW_BURNER_STARTS: int(fields[29]),
+                v.DATA_TOTAL_BURNER_HOURS: int(fields[30]),
+                v.DATA_CH_PUMP_HOURS: int(fields[31]),
+                v.DATA_DHW_PUMP_HOURS: int(fields[32]),
+                v.DATA_DHW_BURNER_HOURS: int(fields[33]),
+            }
+        else:
+            capmodlimits = fields[4].split("/")
+            dhw_setp_bounds = fields[13].split("/")
+            ch_setp_bounds = fields[14].split("/")
+            status = {
+                v.DATA_SLAVE_MAX_RELATIVE_MOD: float(fields[3]),
+                v.DATA_ROOM_SETPOINT: float(fields[5]),
+                v.DATA_REL_MOD_LEVEL: float(fields[6]),
+                v.DATA_CH_WATER_PRESS: float(fields[7]),
+                v.DATA_ROOM_TEMP: float(fields[8]),
+                v.DATA_CH_WATER_TEMP: float(fields[9]),
+                v.DATA_DHW_TEMP: float(fields[10]),
+                v.DATA_OUTSIDE_TEMP: float(fields[11]),
+                v.DATA_RETURN_WATER_TEMP: float(fields[12]),
+                v.DATA_DHW_SETPOINT: float(fields[15]),
+                v.DATA_MAX_CH_SETPOINT: float(fields[16]),
+                v.DATA_TOTAL_BURNER_STARTS: int(fields[17]),
+                v.DATA_CH_PUMP_STARTS: int(fields[18]),
+                v.DATA_DHW_PUMP_STARTS: int(fields[19]),
+                v.DATA_DHW_BURNER_STARTS: int(fields[20]),
+                v.DATA_TOTAL_BURNER_HOURS: int(fields[21]),
+                v.DATA_CH_PUMP_HOURS: int(fields[22]),
+                v.DATA_DHW_PUMP_HOURS: int(fields[23]),
+                v.DATA_DHW_BURNER_HOURS: int(fields[24]),
+            }
+        status.update(
```

Comment:
For future maintainability of the code it may be better to completely separate the dict-building process between v4 and v5, i.e. don't use status.update() but instead create the full dict in both cases above. This may duplicate some assignments, but we will never execute both and it will make it easier to read and find possible errors.